File: fitplanner-app/src/app/modules/user/interface/food-item-creation-request.ts
import { FoodItem } from "./food-item";

export interface FoodItemCreationRequest {
    email: string,
    date: string,
    mealName: string,
    foodItem: {
        name: string,
        calories: number,
        protein: number,
        fat: number,
        carbs: number,
        quantity: number
    }
}

File: fitplanner-services/nutrition-service/src/main/java/com/fitplanner/nutrition/model/api/FoodItemCreationRequest.java
package com.fitplanner.nutrition.model.api;

import com.fitplanner.nutrition.model.food.FoodItem;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

public record FoodItemCreationRequest(
        @NotNull @NotEmpty String email,
        @NotNull @NotEmpty String date,
        @NotNull @NotEmpty String mealName,
        @NotNull FoodItem foodItem,
        @NotNull Integer calories,
        @NotNull Double protein,
        @NotNull Double fat,
        @NotNull Double carbs
) {}